crates/core/component/shielded-pool/src/nullifier_derivation.rs

```diff
@@ -60,12 +61,12 @@ impl ConstraintSynthesizer<Fq> for NullifierDerivationCircuit {
 
         // Public inputs
         let claimed_nullifier_var = NullifierVar::new_input(cs.clone(), || Ok(self.nullifier))?;
-        let note_var = note::NoteVar::new_input(cs.clone(), || Ok(self.note.clone()))?;
+        let note_commitment_var =
```

Comment:
One other change we should make as part of this is below in `NullifierDerivationProof::verify` we'll want to pass in the note commitment instead of the `Note` as an argument, and then when we construct the public inputs the second element in the public input vector ([here](https://github.com/penumbra-zone/penumbra/pull/2833/files#diff-f4e7e8f2b1ba9fb55a1dbdc472f800bb724bc65e810b56db8a9f4db2a13c1d4fL150)) should be the note commitment. Otherwise this LGTM!

Reply:
Fixes in the latest commits 8^)